pkg/git/git_keychain.go

```diff
@@ -15,41 +19,98 @@ type GitKeychain interface {
 }
 
 type secretGitKeychain struct {
-	gitCredentials []gitCredentials
-	volumeName     string
+	httpsCredentials []gitCredentials
+	sshCredentials   []gitCredentials
+	volumeName       string
 }
 
 type gitCredentials struct {
 	Domain     string
 	SecretName string
 }
 
-func NewMountedSecretGitKeychain(volumeName string, secrets []string) (*secretGitKeychain, error) {
-	var gitCreds []gitCredentials
-	for _, s := range secrets {
+func NewMountedSecretGitKeychain(volumeName string, basicAuthSecrets, sshAuthSecrets []string) (*secretGitKeychain, error) {
+	var httpsCreds []gitCredentials
+	var sshCredentials []gitCredentials
+
+	for _, s := range basicAuthSecrets {
 		splitSecret := strings.Split(s, "=")
 		if len(splitSecret) != 2 {
 			return nil, errors.Errorf("could not parse git secret argument %s", s)
 		}
 
-		gitCreds = append(gitCreds, gitCredentials{
+		httpsCreds = append(httpsCreds, gitCredentials{
 			Domain:     splitSecret[1],
 			SecretName: splitSecret[0],
 		})
 	}
+	for _, s := range sshAuthSecrets {
+		splitSecret := strings.Split(s, "=")
+		if len(splitSecret) != 2 {
+			return nil, errors.Errorf("could not parse git secret argument %s", s)
+		}
+
+		sshCredentials = append(sshCredentials, gitCredentials{
+			Domain:     splitSecret[1],
+			SecretName: splitSecret[0],
+		})
+	}
+
 	return &secretGitKeychain{
-		gitCredentials: gitCreds,
-		volumeName:     volumeName,
+		httpsCredentials: httpsCreds,
+		sshCredentials:   sshCredentials,
+		volumeName:       volumeName,
 	}, nil
 }
 
 func (k *secretGitKeychain) Resolve(url string) (transport.AuthMethod, error) {
-	for _, creds := range k.gitCredentials {
-		if gitUrlMatch(url, creds.Domain) {
-			basicAuth, err := secret.ReadSecret(k.volumeName, creds.SecretName)
+	var matchedCreds []gitCredentials
+	endpoint, err := transport.NewEndpoint(url)
+	if err != nil {
+		return nil, err
+	}
+	logger := log.New(os.Stdout, "prepare:", log.Lshortfile)
```

Comment:
We should pass the logger into NewMountedSecretGitKeychain most likely.

Reply:
Yeah. I agree. I would like to see if we can pass the logger around to the other resolvers in case we need to just to have prepare behave similarly across the different sources.